libstackerdb/src/libstackerdb.rs

```diff
@@ -0,0 +1,289 @@
+// Copyright (C) 2013-2020 Blockstack PBC, a public benefit corporation
+// Copyright (C) 2020-2023 Stacks Open Internet Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+extern crate clarity;
+extern crate serde;
+extern crate sha2;
+extern crate stacks_common;
+
+use std::error;
+use std::fmt;
+use std::io::{Read, Write};
+
+use sha2::{Digest, Sha512_256};
+
+use stacks_common::types::chainstate::{StacksAddress, StacksPrivateKey, StacksPublicKey};
+use stacks_common::types::PrivateKey;
+use stacks_common::util::hash::{hex_bytes, to_hex, Hash160, Sha512Trunc256Sum};
+
+use stacks_common::codec::read_next;
+use stacks_common::codec::read_next_at_most;
+use stacks_common::codec::write_next;
+use stacks_common::codec::Error as codec_error;
```

Comment:
I know this happens elsewhere, but I think it's a good idea to start renaming `use` structs in CamelCase rather than snake_case -- all of the structs in the codebase are CamelCase (as rust encourages), so when they get imported, it makes sense for them to remain CamelCase.

```suggestion
use stacks_common::codec::Error as CodecError;
```

Reply:
Sure